src/Sulu/Bundle/CustomUrlBundle/DependencyInjection/CompilerPass/PathSegmentCompilerPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CustomUrlBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Add path segments for custom-urls.
+ */
+class PathSegmentCompilerPass implements CompilerPassInterface
+{
+    const PATH_SEGMENT_REGISTRY_ID = 'sulu_document_manager.path_segment_registry';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition(self::PATH_SEGMENT_REGISTRY_ID)) {
+            return;
+        }
+
+        $definition = $container->getDefinition(self::PATH_SEGMENT_REGISTRY_ID);
+        $definition->addMethodCall('add', ['custom-urls', 'custom-urls']);
+        $definition->addMethodCall('add', ['custom-urls-items', 'items']);
+        $definition->addMethodCall('add', ['custom-urls-routes', 'routes']);
```

Comment:
Wouldn't a `PrependExtension` be a better place to do this? Then you would also not need the new lines in https://github.com/sulu-io/sulu-document-manager/pull/58/files#diff-aaf1b8253014c99a5f92c8b8de4d9515R43